example/integration_test/bucketeer_flutter_sdk_test.dart

```diff
@@ -0,0 +1,298 @@
+        ),
+      );
+    });
+
+    testWidgets('testSwitchUser', (WidgetTester _) async {
+      await BKTClient.instance.destroy().onError((error, stackTrace) => fail(
+          "BKTClient.instance.destroy should success and should not throw exception"));
+      final config = BKTConfigBuilder()
+          .apiKey(Constants.API_KEY)
+          .apiEndpoint(Constants.API_ENDPOINT)
+          .featureTag(FEATURE_TAG)
+          .debugging(DEBUGGING)
+          .eventsMaxQueueSize(Constants.EXAMPLE_EVENT_MAX_QUEUE_SIZE)
+          .eventsFlushInterval(Constants.EXAMPLE_EVENTS_FLUSH_INTERVAL)
+          .pollingInterval(Constants.EXAMPLE_POLLING_INTERVAL)
+          .backgroundPollingInterval(
+              Constants.EXAMPLE_BACKGROUND_POLLING_INTERVAL)
+          .appVersion(APP_VERSION)
+          .build();
+      final user = BKTUserBuilder().id("test_id").data({}).build();
+
+      var instanceResult = await BKTClient.initialize(
+        config: config,
+        user: user,
+      );
+      expect(instanceResult.isSuccess, true,
+          reason: "initialize() should success");
+
+      await BKTClient.instance.updateUserAttributes(
+        userAttributes: {'app_version': APP_VERSION},
+      ).onError(
+        (error, stackTrace) => fail(
+            "BKTClient.instance.updateUserAttributes should success and should not throw exception"),
+      );
+
+      var currentUser = await BKTClient.instance.currentUser();
+      expect(currentUser != null, true,
+          reason:
+              "BKTClient.instance.currentUser() should return non-null user data");
+      expect(currentUser!.id, "test_id", reason: "user_id should be `test_id`");
+      expect(currentUser.data, {'app_version': APP_VERSION},
+          reason: "user_data should match");
+
+      var fetchEvaluationsResult =
+          await BKTClient.instance.fetchEvaluations(timeoutMillis: 30000);
+      expect(fetchEvaluationsResult.isSuccess, true,
+          reason: "fetchEvaluations() should success");
+    });
+  }
+
+  group('Bucketeer', () {
+    setUp(() async {
+      final config = BKTConfigBuilder()
+          .apiKey(Constants.API_KEY)
+          .apiEndpoint(Constants.API_ENDPOINT)
+          .featureTag(FEATURE_TAG)
+          .debugging(DEBUGGING)
+          .eventsMaxQueueSize(Constants.EXAMPLE_EVENT_MAX_QUEUE_SIZE)
+          .eventsFlushInterval(Constants.EXAMPLE_EVENTS_FLUSH_INTERVAL)
+          .pollingInterval(Constants.EXAMPLE_POLLING_INTERVAL)
+          .backgroundPollingInterval(
+              Constants.EXAMPLE_BACKGROUND_POLLING_INTERVAL)
+          .appVersion(APP_VERSION)
+          .build();
+      final user = BKTUserBuilder().id(USER_ID).data({}).build();
+
+      var result = await BKTClient.initialize(
+        config: config,
+        user: user,
+      );
+      expect(result.isSuccess, true, reason: "initialize() should success");
+
+      BKTClient.instance.addEvaluationUpdateListener(listener);
```

Comment:
Can a test be added to check if the `OnUpdate` callback is called when the evaluations are updated?

Reply:
@cre8ivejp I have a check here 
https://github.com/duyhungtnn/bkt-flutter/pull/1/files/cc8138f6bac3878ed383775f0c954c4b1bf88a2d#diff-19c7758065b22b6514e5d67c4c9421f3a4d7e11b4e1bcc77f661b9188ef3f869R290

Reply:
let me think if we should add more test on this

Reply:
I added a test to make sure the `listener.onUpdate()` is called before continuing with any other test case.
See the detail 
https://github.com/duyhungtnn/bkt-flutter/pull/1/files#diff-19c7758065b22b6514e5d67c4c9421f3a4d7e11b4e1bcc77f661b9188ef3f869R266

Reply:
moved the `listener` check to a new test case 
https://github.com/duyhungtnn/bkt-flutter/pull/1/files#diff-19c7758065b22b6514e5d67c4c9421f3a4d7e11b4e1bcc77f661b9188ef3f869R67